pkg/occlient/occlient.go

```diff
@@ -533,16 +533,59 @@ func (c *Client) CreateNewProject(projectName string, wait bool) error {
 			if !ok {
 				break
 			}
-			if e, ok := val.Object.(*projectv1.Project); ok {
-				glog.V(4).Infof("Project %s now exists", e.Name)
-				return nil
+			if prj, ok := val.Object.(*projectv1.Project); ok {
+				glog.V(4).Infof("Status of creation of project %s is %s", prj.Name, prj.Status.Phase)
+				switch prj.Status.Phase {
+				//prj.Status.Phase can only be "Terminating" or "Active" or ""
+				case corev1.NamespaceActive:
+					if val.Type == watch.Added {
+						glog.V(4).Infof("Project %s now exists", prj.Name)
+						return nil
+					}
+					if val.Type == watch.Error {
+						return fmt.Errorf("failed watching the deletion of project %s", prj.Name)
+					}
+				}
 			}
 		}
 	}
 
 	return nil
 }
 
+// WaitForServiceAccountInNamespace waits for the given service account to be ready
+func (c *Client) WaitForServiceAccountInNamespace(namespace, serviceAccountName string) error {
+	if namespace == "" || serviceAccountName == "" {
+		return errors.New("namespace and serviceAccountName cannot be empty")
+	}
+	watcher, err := c.kubeClient.CoreV1().ServiceAccounts(namespace).Watch(metav1.SingleObject(metav1.ObjectMeta{Name: serviceAccountName}))
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	timeout := time.After(1 * time.Minute)
+	if watcher != nil {
```

Comment:
shouldn't this be above https://github.com/openshift/odo/pull/2330/files#diff-6025e9b620e9321c03c11bf29e5c4dbdR565
because else we are calling `Stop()` on a `nil` watcher?

Reply:
or is this check even necessary?